wkcuber/tile_cubing.py

```diff
@@ -1,60 +1,194 @@
+
+
+def replace_coordinates(
+    pattern: str, coord_ids_with_replacement_info: Dict[str, Tuple[int, int]]
+) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids_with_replacement_info:
+            number_of_digits = coord_ids_with_replacement_info[coord][1]
+            format_str = "0" + str(number_of_digits) + "d"
+            pattern = pattern.replace(
+                occurrence,
+                format(coord_ids_with_replacement_info[coord][0], format_str),
+                1,
+            )
+    return pattern
+
+
+def replace_pattern_to_specific_length_without_brackets(
+    pattern: str, coord_ids_with_specific_length: Dict[str, int]
+) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids_with_specific_length:
+            pattern = pattern.replace(
+                occurrence, coord * coord_ids_with_specific_length[coord], 1
+            )
+    return pattern
 
-def find_source_sections(source_path):
-    section_folders = [
-        f for f in listdir(source_path) if path.isdir(path.join(source_path, f))
-    ]
-    section_folders = [path.join(source_path, s) for s in section_folders]
-    section_folders.sort()
-    return section_folders
 
+def replace_coordinates_with_glob_regex(pattern: str, coord_ids: Dict[str, int]) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids:
+            number_of_digits = coord_ids[coord]
+            pattern = pattern.replace(occurrence, "[0-9]" * number_of_digits, 1)
+    return pattern
 
-def parse_tile_file_name(filename):
-    m = CUBE_REGEX.search(filename)
-    if m is None:
-        return None
-    return (int(m.group(1)), int(m.group(2)), m.group(3))
 
+""" Counts how many digits the dimensions x, y and z occupy in the given pattern. """
 
-def find_source_files(source_section_path):
-    all_source_files = [
-        path.join(source_section_path, s)
-        for s in find_files(
-            path.join(source_section_path, "**", "*"), image_reader.readers.keys()
+
+def get_digit_counts_for_dimensions(pattern):
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    decimal_lengths = {"x": 0, "y": 0, "z": 0}
+
+    for occurrence in occurrences:
+        current_dimension = occurrence[1]
+        decimal_lengths[current_dimension] = max(
+            decimal_lengths[current_dimension], len(occurrence) - 2
         )
-    ]
 
-    all_source_files.sort()
-    return all_source_files
+    return decimal_lengths
+
+
+def detect_interval_for_dimensions(
+    file_path_pattern: str, decimal_lengths: Dict[str, int]
+) -> Tuple[Dict[str, int], Dict[str, int], str, int]:
+    arbitrary_file = None
```

Comment:
This variable is not used i I see it correctly? You could assign to this variable for the first value of `file_name` and return `arbitrary_file` instead of `file_name`. In my opinion, it's a bit weird to return the loop variable `file_name` outside of the loop (in other languages, this wouldn't be possible since the variable is often only defined in the loop). But that's my personal take on that. Either remove the above variable or change it as described :)